### DISCUSSION TITLE
[远期] 单词表

### DISCUSSION THREAD

**Discussion by Guanchishan** [MindongLab/yngdieng#20] [产品功能 ideas]
**你提出的功能想解决什么问题**

1. 按特定主题收藏特定内容，以供未来学习。

**你觉得理想的解决方案是什么**

1. 参考Moji日语，允许用户建立多个单词表
1. 虽然说是单词表，但也允许用户收藏“[句子](https://github.com/MindongLab/cdo-corpus/blob/master/sentences.csv)”（如单词例句）与“[作品](https://github.com/MindongLab/cdo-corpus/blob/master/opera/Min%20opera/%5Bcdo-hani%5D%E6%98%A5%E8%8D%89%E9%97%96%E5%A0%82%20%E6%89%9B%E8%BD%8E%E7%89%87%E6%AE%B5.srt)”等（Moji日语在这方面做的更激进，甚至允许透过内置浏览器收藏网页，与其他收藏的词汇并列；但没看出榕典有学到这个程度的必要，因为网上的福州话网页没想象中那么多；但是讲述福州词汇与掌故的博客与微信文章以及闽东维基自身的条目数还是颇有存量的所以也并不是不可以考虑）
1. 允许文件夹树，即大单词表套小单词表
1. 允许用户共享自己的单词表（即建立共享平台），并允许其他用户收藏

**其他可能的解决方案**
- 可能用户可以自己想办法记memo，但能提供一站式服务的话肯定更能促进使用，增强用户粘性。

- 作为“竞品”的闽江学院福州话app也有收藏夹功能，可收藏单字，但不支援多个单词表，也不支援文件夹树。

**其他信息**
- 开放共享后可能存在内容审核问题。考虑到榕典并不是大产品，就采取先允许共享后接受意见反馈的维基模式吧。

**Comment by Guanchishan** [comment]
按 #88 的讨论：

> DDFShanyu commented：
contrib里面的tag我觉得意义不大反而，因为怎么去给词汇分类，没有统一标准，这样会导致分类很乱。

- [x] 可能计划索性将tag改造为符合冯版要求的分类；

- [x] 而目前tag内如“福州公交八路”之类，可能计划建一个flashcard.???之类的文件收纳之。

- 目前只是在cdo-corpus的sentences以及foochow-idiom-data的other idioms里建立了flashcard字段。

目前将单词表信息从contrib.tsv中抽出暂录于此：

- [x] 词表标题：福州公交八路

- 于山
- 正大廣場
- 光明橋
- 狀元境
- 貿總酒店

顺带一提，

> 作为“竞品”的闽江学院福州话app也有收藏夹功能

闽院app似乎不能用了。

**Comment by Guanchishan** [comment]
flashcard表可以允许不同的用户去star，也可以允许不同的用户自己folk一份落来，未来再push给第一作者什么的。这样设计为了满足以下要求：

1. 可能原作者一个不开心，或者一个误操作，把表单删了，但是明明表单很有价值大家都想用；
1. folk这一行为本身也符合前电子化时代人们买并传阅单词本、抄学霸单词本、看完别人单词本后自己加一两个词等等的使用习惯；
1. folk得到的表最好能显示原作者，并提供跳转回原表的链接——这样就和GitHub一样。
1. 有的人如果自己没有组织词汇的能力，只想等着别人实时更新单词表，那他完全可以不folk只star。

**Comment by ztl8702** [comment]
单词表还有一个“妙用”，像 真鸟公众号 的推送，其实可以为每一期创建一个单词表 --- "想复习本期词汇，上榕典“

**Comment by Guanchishan** [comment]
对，甚至比如可以：
每次有推送都放到专门的公开单词表，如“每日一词@2019-06-04”，然后又去属于比如“每日一词”这样的公开母表单（这需要单词表树功能）
许多词典软件或网页都有每日一词专栏，还能查询过往专栏内容，榕典也可以搞个，未来把存稿都放上去

发自我的iPhone

**Comment by Guanchishan** [comment]
设计稿@2020-08-17
![image](https://user-images.githubusercontent.com/30859354/90687381-590e2c80-e29f-11ea-90dc-99405d596ede.png)
![image](https://user-images.githubusercontent.com/30859354/90687393-5c091d00-e29f-11ea-912b-93991545e788.png)
右上角的两张📜相叠之图标代表folk功能。该功能与普通的“收藏别人的词表”不一样，之前的讨论见[此楼](https://github.com/MindongLab/yngdieng/issues/20#issuecomment-661111275)。我们可以把该功能称为“誊”，添点虾油味。
![image](https://user-images.githubusercontent.com/30859354/90687733-e0f43680-e29f-11ea-8af3-5b35fc62d3db.png)
